tests/likelihood/test_likelihood.py

```diff
@@ -19,7 +19,7 @@ def compute_loglike(self, _: like.ModelingTools) -> float:
 class LikelihoodThatThrowsUnhandledError(like.Likelihood):
     """A likelihood that always throws a pyccl integration error exception."""
```

Comment:
Not directly connected to this PR but these tests confused me initially. Updating the doc string and maybe a comment on what's going on (I guess integration errors get ignored by setting the log like to -inf but other errors propagate?) would be useful I think.